docs/autorenew-feature.md

```diff
@@ -0,0 +1,71 @@
+# The autorenew feature
+
+## Goal
+-	Allow Hedera Services to automatically search for entities such as accounts, files, smart contracts, topics, tokens... that are about to expire and renew them.
+
+## Design
+- Introduce new settings in `application.properties`:
+  * `autorenew.isEnabled`
+  * `autorenew.numberOfEntitiesToCheck`
+  * `autorenew.maxNumberOfEntitiesToRenew`
+  * `autorenew.gracePeriod`
+- Each Hedera entity has an `expirationTime` which is the effective consensus timestamp at (and after) which the entity is set to expire.
+- Each Hedera entity also has an `autoRenewAccount` which is the account to pay for the fee at renewal. This `autoRenewAccount` could be itself if the entity is a crypto account or an account associated with the entity when it was created.
+- When a Hedera entity is created, its initial lifetime is defined by its `autoRenewPeriod`. At its `expirationTime`, Hedera Services will try to extend an entity's lifetime by another `autoRenewPeriod` if the `autoRenewAccount` has enough balance to do so, or as much extension as the remaining balance permits.
```

Comment:
Would there be a way to control the auto renew feature for any entity when that entity is created, for eg a user may want an account to auto renew but not a specific file they created.

Reply:
We don't control the ownership of an entity in the system. Technically, if you have the (admin) key of the entity, you can update or delete the entity.

Reply:
For a user who created an entity, would they be able to control the auto renewal of that entity at the time of creation or later update to that entity?